src/main/java/mate/academy/mybookshop/service/UserServiceImpl.java

```diff
@@ -17,19 +18,25 @@
 @Service
 public class UserServiceImpl implements UserService {
     private final UserRepository userRepository;
+    private final RoleRepository roleRepository;
     private final UserMapper userMapper;
     private final PasswordEncoder passwordEncoder;
 
+    private long userRoleId;
+
+    @PostConstruct
+    void init() {
+        userRoleId = roleRepository.findByType(RoleEntity.RoleType.USER).getId();
+    }
+
```

Comment:
Please see Oksana Miazina comments, look like you copipasted each other

Reply:
@andrii-hoienko Actually, I stole it from Oleksiy, but he let me)

Reply:
in this way we avoid the need to go to the database every time to find the role ID, because of this we can use `getReferenceById()`

Reply:
Agree with the comment there:
> Not sure that this should be state of service, maybe it makes sense to make some class like UserConfigHolder and put it there to separate service and some config responsibilities

in general, it seems like a good idea. Still, then we are essentially hardcoding the id of that role that we have got throughout the whole life of the application. And yes, I'd also suggest moving it to some custom class like `RoleContextUtil` where you would initialize and store this data